### PR TITLE
feat(cli): Optimise serving static assets

### PR DESCRIPTION
**Pull request by netroy**
N8N-4739
N8N-4805

### CODE REVIEW

File: packages/cli/src/Server.ts [hunk 1]
```diff
@@ -1801,42 +1766,41 @@ class App {
...
+								.replace(firstLinkedScriptSegment, scriptsString + firstLinkedScriptSegment);
 						}
-					},
-				}),
+						const destFile = pathJoin(generatedStaticDir, fileName);
+						await mkdir(pathDirname(destFile), { recursive: true });
+						await writeFile(destFile, payload, 'utf-8');
+						send(req, destFile).pipe(res);
+						return;
+					}
```
Comment (alexgrozav) [inline packages/cli/src/Server.ts:1790]: Would it make sense to generate the pre-processed `public` folder beforehand and just serve assets from the pre-processed dir?

This way we could eliminate these checks on every asset request.

Comment (netroy) [reply packages/cli/src/Server.ts:1790]: good catch. I'll try to give it a shot.

Comment (netroy) [reply packages/cli/src/Server.ts]: Done :heavy_check_mark: 
I also discovered today that we are not returning `Cache-Control` headers on these static assets on master ([Ticket](https://linear.app/n8n/issue/N8N-4805)). Switching to pre-generating these and using `express.static` fixed that as well.

**Review by alexgrozav — APPROVED** [review]
Amazing work, @netroy. 🙌

**Comment by janober** [comment]
Got released with `n8n@0.196.0`

**Pull request merged**